otcextensions/sdk/dis/v2/app.py

```diff
@@ -0,0 +1,137 @@
+
+    allow_create = True
+    allow_list = True
+    allow_delete = True
+    allow_fetch = True
+
+    _query_mapping = resource.QueryParameters(
+        'limit', 'start_app_name', 'stream_name'
+    )
+
+    # Properties
+    #: Unique identifier of the app.
+    app_id = resource.Body('app_id', alternate_id=True)
+    #: List of associated streams.
+    commit_checkpoint_stream_names = \
+        resource.Body('commit_checkpoint_stream_names', type=list)
+    #: Timestamp at which the DIS app was created.
+    created_at = resource.Body('create_time', type=int)
+    #: App name.
+    name = resource.Body('app_name')
+    #: Stream Name
+    stream_name = resource.Body('stream_name')
+
+    @classmethod
+    def list(cls, session, paginated=True, base_path=None,
+             allow_unknown_params=False, **params):
+
+        if not cls.allow_list:
+            raise exceptions.MethodNotSupported(cls, "list")
+        session = cls._get_session(session)
+        microversion = cls._get_microversion(session, action='list')
+
+        if base_path is None:
+            base_path = cls.base_path
+        params = cls._query_mapping._validate(
+            params, base_path=base_path,
+            allow_unknown_params=allow_unknown_params)
+
+        query_params = cls._query_mapping._transpose(params, cls)
+        uri = base_path % params
+
+        while uri:
+            # Copy query_params due to weird mock unittest interactions
+            response = session.get(
+                uri,
+                headers={"Accept": "application/json"},
+                params=query_params.copy(),
+                microversion=microversion)
+            exceptions.raise_from_response(response)
+            data = response.json()
+
+            if cls.resources_key:
+                resources = data[cls.resources_key]
+            else:
+                resources = data
+
+            if not isinstance(resources, list):
+                resources = [resources]
+
+            for raw_resource in resources:
+                value = cls.existing(
+                    microversion=microversion,
+                    connection=session._get_connection(),
+                    **raw_resource)
+                yield value
+
+            if resources and paginated:
+                uri, next_params = cls._get_next_link(
+                    uri, response, data)
+                query_params.update(next_params)
+            else:
+                return
+
+    @classmethod
+    def _get_next_link(cls, uri, response, data):
+        # RDS service pagination. Returns query for the next page
+        next_link = None
+        params = {}
+        if 'has_more_app' in data and data['has_more_app']:
+            next_link = uri
+            params['start_app_name'] = data['apps'][-1]['app_name']
+        else:
+            next_link = None
+        query_params = cls._query_mapping._transpose(params, cls)
+        return next_link, query_params
```

Comment:
look comment at data.py

Reply:
removed pagination logic from data.py , in DIS every sub service list API has different pagination params, so we cannot bring to _base.py